underfs/s3/src/test/java/tachyon/underfs/s3/S3UnderFileSystemTest.java

```diff
@@ -16,23 +16,131 @@
+import org.powermock.reflect.Whitebox;
 
 /**
- * This test ensures the s3 ufs module correctly accepts paths that begin with s3n://
+ * Tests for the private helper methods in {@link tachyon.underfs.s3.S3UnderFileSystem} that do
+ * not require an S3 backend
  */
 public class S3UnderFileSystemTest {
+  private static S3UnderFileSystem sMockS3UnderFileSystem;
+
+  @BeforeClass
+  public static final void beforeClass() {
+    sMockS3UnderFileSystem = PowerMockito.mock(S3UnderFileSystem.class);
+    Whitebox.setInternalState(sMockS3UnderFileSystem, "mBucketName", "test-bucket");
+    Whitebox.setInternalState(sMockS3UnderFileSystem, "mBucketPrefix", "s3n://test-bucket/");
+  }
+
   @Test
-  public void factoryTest() {
-    TachyonConf conf = new TachyonConf();
+  public void convertToFolderNameTest() throws Exception {
+    String input1 = "test";
+    String result1 = Whitebox.invokeMethod(sMockS3UnderFileSystem, "convertToFolderName", input1);
 
-    UnderFileSystemFactory factory =
-        UnderFileSystemRegistry.find("s3n://localhost/test/path", conf);
-    Assert.assertNotNull(
-        "A UnderFileSystemFactory should exist for s3n paths when using this module", factory);
+    Assert.assertEquals(result1, "test_$folder$");
   }
+
+  @Test
+  public void getChildNameTest() throws Exception {
+    String input11 = "s3n://test-bucket/child";
+    String input12 = "s3n://test-bucket/";
+    String input21 = "s3n://test-bucket/parent/child";
+    String input22 = "s3n://test-bucket/parent/";
+    String input31 = "s3n://test-bucket/child";
+    String input32 = "s3n://test-bucket/not-parent";
+    String result1 =
+        Whitebox.invokeMethod(sMockS3UnderFileSystem, "getChildName", input11, input12);
+    String result2 =
+        Whitebox.invokeMethod(sMockS3UnderFileSystem, "getChildName", input21, input22);
+    String result3 =
+        Whitebox.invokeMethod(sMockS3UnderFileSystem, "getChildName", input31, input32);
+
+    Assert.assertEquals("child", result1);
+    Assert.assertEquals("child", result2);
+    Assert.assertEquals("s3n://test-bucket/child", result3);
+  }
+
+  @Test
+  public void getParentKeyTest() throws Exception {
+    String input1 = "s3n://test-bucket/parent-is-root";
+    String input2 = "s3n://test-bucket/";
+    String input3 = "s3n://test-bucket/parent/child";
+    String input4 = "s3n://test-bucket";
+    String result1 = Whitebox.invokeMethod(sMockS3UnderFileSystem, "getParentKey", input1);
+    String result2 = Whitebox.invokeMethod(sMockS3UnderFileSystem, "getParentKey", input2);
+    String result3 = Whitebox.invokeMethod(sMockS3UnderFileSystem, "getParentKey", input3);
+    String result4 = Whitebox.invokeMethod(sMockS3UnderFileSystem, "getParentKey", input4);
+
+    Assert.assertEquals("s3n://test-bucket", result1);
+    Assert.assertEquals(null, result2);
+    Assert.assertEquals("s3n://test-bucket/parent", result3);
+    Assert.assertEquals(null, result4);
+  }
+
```

Comment:
Oh, that's right. Perhaps in future PRs, we could considering injecting mock S3. But in this PR, I think it is good :)